api/src/main/java/com/graphhopper/GraphHopperConfig.java

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.graphhopper;
+
+import com.graphhopper.util.PMap;
+
+import java.util.Map;
+
+/**
+ * This class represents the global configuration for the GraphHopper class, which is typically configured via the
+ * `config.yml` file. So far we are mapping the key-value pairs in the config file to a string-string map, but soon
+ * we will start adding hierarchical configurations (lists, nested objects etc.). We will also start adding the
+ * different configuration options as fields of this class including the default values.
+ */
+public class GraphHopperConfig {
+    private final PMap map;
+
+    public GraphHopperConfig() {
+        this(new PMap());
+    }
+
+    public GraphHopperConfig(PMap pMap) {
+        this.map = pMap;
+    }
+
+    public GraphHopperConfig put(String key, Object value) {
+        map.put(key, value);
+        return this;
+    }
+
+    public boolean has(String key) {
+        return map.has(key);
+    }
+
+    public boolean getBool(String key, boolean _default) {
+        return map.getBool(key, _default);
+    }
+
+    public int getInt(String key, int _default) {
+        return map.getInt(key, _default);
+    }
+
+    public long getLong(String key, long _default) {
+        return map.getLong(key, _default);
+    }
+
+    public float getFloat(String key, float _default) {
+        return map.getFloat(key, _default);
+    }
+
+    public double getDouble(String key, double _default) {
+        return map.getDouble(key, _default);
+    }
+
+    public String get(String key, String _default) {
+        return map.get(key, _default);
+    }
+
+    public PMap asPMap() {
+        return map;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("properties:\n");
+        for (Map.Entry<String, String> entry : map.toMap().entrySet()) {
+            sb.append(entry.getKey()).append(": ").append(entry.getValue());
+            sb.append("\n");
+        }
+        return sb.toString();
```

Comment:
This is now a bit too ugly as without indent and looks like it should be debugging info IMO:

```
2020-02-14 19:47:56.845 [main] INFO  c.g... properties:
graph.dataaccess: RAM_STORE
ch.enabled: false
graph.flag_encoders: car
prepare.ch.weightings: fastest
prepare.min_one_way_network_size: 0
graph.location: graph-cache
prepare.min_network_size: 0
routing.ch.disabling_allowed: true
routing.max_visited_nodes: 1500000
prepare.lm.weightings: fastest
routing.round_trip.max_retries: 2
```

Reply:
Yes `toString` is for debugging or what do you mean? The output should be indented another way so it resembles the yaml file?

Reply:
Currently we print it when it starts here: https://github.com/graphhopper/matrix/blob/master/matrix/src/main/java/com/graphhopper/matrix/http/GraphHopperEEManaged.java#L95

Ah, maybe I just change it to:

`logger.info(ghConfig.asPMap().toString());`

?

Reply:
We can adjust `toString()` to whatever you/we think is useful. What do you not like about the current format? Do you think its too verbose (because of the new lines), or would you prefer `key=value` over `key: value`? Note that as soon as we add profiles to the config (or other typed parameters `asPMap().toString()` will no longer contain the whole information.

Reply:
Yes, the new line is good for debugging but a bit too much for normal startup info.

> Note that as soon as we add profiles to the config (or other typed parameters asPMap().toString() will no longer contain the whole information.

Yes, for now I just need to make sure certain info is correctly picked up

Reply:
Ah is it? I thought this is something quite useful at startup. But yeah feel free to change it to whatever you prefer, I do not care that much.